prov/rxm/src/rxm.h

```diff
@@ -280,18 +281,26 @@ struct rxm_mr {
 	struct rxm_domain *domain;
 };
 
-struct rxm_ep_wire_proto {
-	uint8_t	ctrl_version;
-	uint8_t	op_version;
-	uint8_t endianness;
-	uint8_t padding[5];
-	uint64_t eager_size;
-};
-
-struct rxm_cm_data {
-	struct sockaddr name;
-	uint64_t conn_id;
-	struct rxm_ep_wire_proto proto;
+union rxm_cm_data {
```

Comment:
I wonder if 3 separate structures wouldn't be cleaner than forming this unoion

Reply:
I initially had the version field common and as part of the union directly but later moved it inside "connect struct" since that's where we need to check for version. Still I kept the three struct together as that avoids defining separate ones.

Now I think the version field may be needed in "reject struct". In case we change reject struct in future, the peer who's processing the reject must know the version to correctly process the fields. The reject could be in fact due to a mismatched version.